invenio_vocabularies/administration/views/vocabularies.py

```diff
@@ -54,45 +57,64 @@ def get_api_endpoint(self, pid_value=None):
         else:
             return f"/api/vocabularies/{pid_value}"
 
-    def get_context(self, **kwargs):
-        """Create details view context."""
-        search_conf = self.init_search_config(**kwargs)
-        schema = self.get_service_schema()
-        serialized_schema = self._schema_to_json(schema)
+    def init_search_config(self, **kwargs):
+        """Build search view config."""
+        pid_value = kwargs.get("pid_value", "")
+        custom_search_config = current_app.config[self.search_config_name].get(
+            pid_value
+        )
+
+        if custom_search_config:
+            available_sort_options = current_app.config[self.search_sort_config_name]
+            available_facets = current_app.config.get(self.search_facets_config_name)
+
+            return partial(
+                search_app_config,
+                config_name=self.get_search_app_name(**kwargs),
+                available_facets=available_facets,
+                sort_options=available_sort_options,
+                endpoint=self.get_api_endpoint(**kwargs),
+                headers=self.get_search_request_headers(**kwargs),
+                sort=SortConfig(
+                    available_sort_options,
+                    custom_search_config["sort"],
+                    custom_search_config["sort_default"],
+                    custom_search_config["sort_default_no_query"],
+                ),
+                facets=FacetsConfig(available_facets, custom_search_config["facets"]),
+            )
+        else:
+            return super().init_search_config(**kwargs)
+
+    def get(self, **kwargs):
+        """GET view method."""
+        parent_context = super().get_context(**kwargs)
+
         pid_value = kwargs.get("pid_value", "")
-        return {
-            "search_config": search_conf,
-            "title": f"{pid_value.capitalize()} vocabulary items",
-            "name": self.name,
-            "resource_schema": serialized_schema,
-            "fields": self.item_field_list,
-            "display_search": self.display_search,
-            "display_create": self.display_create,
-            "display_edit": self.display_edit,
-            "display_delete": self.display_delete,
-            "display_read": self.display_read,
-            "actions": self.serialize_actions(),
-            "pid_path": self.pid_path,
-            "pid_value": pid_value,
-            "create_ui_endpoint": self.get_create_view_endpoint(),
-            "list_ui_endpoint": self.get_list_view_endpoint(),
-            "resource_name": (
-                self.resource_name if self.resource_name else self.pid_path
-            ),
-        }
+        vocab_admin_cfg = current_app.config["VOCABULARIES_ADMINISTRATION_CONFIG"]
```

Comment:
I would like to suggest using class factory pattern here instead of moving it to a static dict config in the configuration variable. I understand the dict is imitating the AdminView class interface but if it is not really a class, then we diverge of the pattern of implementation of the administration module